### PR TITLE
fix(@desktop/chat): Fix user names propagation.

### LINKED ISSUE #8110: Creating a private group - The name of group is wrong
# Bug Report

## Description

If I create a private group with 2 other people , and I have a nickname for one of the users, 

1. The group name doesnt contain my name and
2. The name of my nickname of the memeber is shown to the other use rather than the nicknmae as saved on the group members app

Screenshot 2022-11-03 at 3 11 48 PM

### Additional Information

- Status desktop version: release 0.7.0
- Operating System: MacOS

### LINKED ISSUE #8343: Members list displays nickname twice
# Bug Report

## Description
Member list delegate should display original name in brackets, see image below:

image

## Steps to reproduce

- nickname some contact
- create a group chat with this contact

#### Expected behavior

- original name is displayed in brackets

#### Actual behavior

- nickname is displayed in brackets

### Additional Information

This is most likely related to: https://github.com/status-im/status-desktop/blob/80bb09cc67b4782631acc5170f797f11a1466ce4/src/app_service/service/contacts/dto/contacts.nim#L180-L184

It makes display name inaccessible from within members model: https://github.com/status-im/status-desktop/blob/80bb09cc67b4782631acc5170f797f11a1466ce4/src/app/modules/main/chat_section/chat_content/users/module.nim#L145-L159

because `memberItem.displayName == localNickname` and `memberItem.localNickname == localNickname`

### PR DESCRIPTION
**Pull request by endulab**
Fix #8343
Fix #8110

### What does the PR do

Add ProfileUtils for displaying correct name.
Set names in models correctly.
Fill names in qml correctly.

### Affected areas

Members list, contacts list, creating chat users list.

### Screenshot of functionality (including design for comparison)

image

image

### CODE REVIEW

File: src/app/modules/main/module.nim [hunk 1]
```diff
@@ -767,13 +767,11 @@ method getVerificationRequestFrom*[T](self: Module[T], publicKey: string): Verif
 
 method getContactDetailsAsJson*[T](self: Module[T], publicKey: string, getVerificationRequest: bool): string =
   let contact = self.controller.getContact(publicKey)
-  let (name, _, _) = self.controller.getContactNameAndImage(contact.id)
   var requestStatus = 0
   if getVerificationRequest:
     requestStatus = self.getVerificationRequestFrom(publicKey).status.int
   let jsonObj = %* {
-    "displayName": name,
-    "optionalName": self.controller.getContactDetails(contact.id).optionalName, # original display name, if renamed
```
Comment (caybro) [inline src/app/modules/main/module.nim:776]: We still need this in the community user list panel, to be able to display the _original_ display name below the local nickname

File: ui/imports/shared/views/ProfileDialogView.qml [hunk 2]
```diff
@@ -51,6 +51,10 @@ Pane {
         readonly property string userDisplayName: contactDetails.displayName
         readonly property string userNickName: contactDetails.localNickname
         readonly property string prettyEnsName: contactDetails.name
+        readonly property string aliasName: contactDetails.alias
```
Comment (caybro) [inline ui/imports/shared/views/ProfileDialogView.qml:54]: This doesn't have to be a property on its own (the only usage is in the 2 private properties below)

File: src/app/modules/main/module.nim (hunk 1 above)
Comment (endulab) [reply src/app/modules/main/module.nim:776]: "optionalName" was used in one place only: ProfileDialogView. I removed usage of getContactDetailsAsJson there so this field is not needed anymore.

Comment (caybro) [reply src/app/modules/main/module.nim:776]: OK, thx

**Review by osmaczko — APPROVED** [review]
Nice!

**Review by caybro — APPROVED** [review]
Looks good!

**Comment by endulab** [comment]
@elina2015 can you check if the names are ok all around the application ? Thanks ! 💐

**Pull request merged**